### PR TITLE
Feature/custom presentation

### LINKED ISSUE #2: Configure how the locale name will be displayed
### Feature proposal

Depending on the purpose of the field, it might make sense to display the locale names in the UI locale. Google Translate follows that approach:
image

In other scenarios, it makes sense to display the locale name in the locale itself (so that users can select their locale, which will be written in their own language). Facebook follows that approach:
image

### Describe solution expectations

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by elPeiretti**
resolves #2

Adds feature that allows the developer to choose the locale that is used to render the items.

### CODE REVIEW

**Comment by elPeiretti** [comment]
fixed commit name with rebase

File: src/test/java/com/flowingcode/vaadin/addons/localecombobox/LocaleComboBoxDemo.java [hunk 1]
```diff
@@ -21,17 +21,71 @@
...
+    koreanLocaleCombo.setDisplayMode(LocaleComboBox.DISPLAY_CUSTOM);
+    koreanLocaleCombo.setValue(Locale.ITALY);
+    
+    selectedLocaleCombo.setDisplayMode(LocaleComboBox.DISPLAY_SELECTED);
+    selectedLocaleCombo.setValue(Locale.ITALY);
+    
+    // #if vaadin eq 0
+    add(createHorizontalContainer("Default display mode (uses default locale):", defaultDisplayLocale),
+        createHorizontalContainer("Display locales with France locale:", franceLocaleCombo),
+        createHorizontalContainer("Display locales with with Germany locale:", germanyLocaleCombo),
```
Comment (mlopezFC) [inline src/test/java/com/flowingcode/vaadin/addons/localecombobox/LocaleComboBoxDemo.java]: Remove extra "with" word

File: src/test/java/com/flowingcode/vaadin/addons/localecombobox/LocaleComboBoxDemo.java [hunk 2]
```diff
@@ -21,17 +21,71 @@
...
+    LocaleComboBox koreanLocaleCombo = new LocaleComboBox();
+    LocaleComboBox selectedLocaleCombo = new LocaleComboBox();
+
+    defaultDisplayLocale.setValue(Locale.ITALY);
+    
+    franceLocaleCombo.setDisplayLocale(Locale.FRANCE);
+    franceLocaleCombo.setDisplayMode(LocaleComboBox.DISPLAY_CUSTOM);
+    franceLocaleCombo.setValue(Locale.ITALY);
+
+    germanyLocaleCombo.setDisplayLocale(Locale.GERMANY);
```
Comment (mlopezFC) [inline src/test/java/com/flowingcode/vaadin/addons/localecombobox/LocaleComboBoxDemo.java]: Maybe with just one LocaleComboBox.DISPLAY_CUSTOM example is enough, let's remove the extra ones

**Review by mlopezFC — CHANGES_REQUESTED** [review]
Just some minor details.

File: src/main/java/com/flowingcode/vaadin/addons/localecombobox/LocaleComboBox.java [hunk 3]
```diff
@@ -42,18 +42,74 @@ public class LocaleComboBox extends ComboBox<Locale> {
   private static final String ITEM_LAYOUT_CLASS_NAME = "fc-locale-combo-box-item-layout";
   private static final String ITEM_FLAG_CLASS_NAME = "fc-locale-combo-box-item-flag";
 
+  /**
+   * Constant for the default display mode.
+   * <p>
+   * In this mode, the Locale names are displayed using the default locale's display format.
+   */
+  public static final int DISPLAY_DEFAULT = 0;
```
Comment (javier-godoy) [inline src/main/java/com/flowingcode/vaadin/addons/localecombobox/LocaleComboBox.java]: Use an enum instead of magic constants.

**Review by javier-godoy — CHANGES_REQUESTED** [review]

File: src/main/java/com/flowingcode/vaadin/addons/localecombobox/LocaleComboBox.java (hunk 3 above)
Comment (elPeiretti) [reply src/main/java/com/flowingcode/vaadin/addons/localecombobox/LocaleComboBox.java]: Changed the constants to a new enum inside the class and also updated the javadoc comments

**Review by javier-godoy — APPROVED** [review]

**Review by mlopezFC — APPROVED** [review]
LGTM!

**Pull request merged**